### PR TITLE
LPS-171913 | Test Automation | Master

### PR DESCRIPTION
**Pull request by davisaints**
* https://issues.liferay.com/browse/LPS-171913

### CODE REVIEW

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 1]
```diff
@@ -271,6 +271,119 @@ definition {
...
 	}
 
+	@description = "LPS-171913 - Verify it's possible to add an object entry with the trigger On Payment Status Update"
+	@priority = "4"
+	test CanCreateActionAddObjectEntryWithOnPaymentStatusUpdate {
+		task ("Given: An object with a field is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object",
+				objectName = "CustomObject",
+				pluralLabelName = "Custom Objects");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:281]: You can use LPS when adding a Custom Object, this way avoid inconsistencies in the CI
```
ObjectAdmin.addObjectViaAPI(
	labelName = "Custom Object 171913",
	objectName = "CustomObject171913",
	pluralLabelName = "Custom Objects 171913");
```

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 2]
```diff
@@ -271,6 +271,119 @@ definition {
...
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "true",
+				objectName = "CustomObject");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:289]: Remember to replace all of them.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 3]
```diff
@@ -271,6 +271,119 @@ definition {
...
+			Button.click(button = "Submit");
+
+			Refresh();
+		}
+
+		task ("Than: The action is triggered") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "Custom Objects");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:381]: You can use macro via API here

`ObjectAdmin.goToCustomObject(objectName = "CustomObject171913");`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase (hunk 2 above)
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:289]: Remember to replace all of them.

Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:289]: Remember to replace all of them.

`objectName = "CustomObject");` to `objectName = "CustomObject171913");`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 4]
```diff
@@ -271,6 +271,119 @@ definition {
...
+			SelectFrame.selectFrameNoLoading(locator1 = "IFrame#MODAL_BODY");
+
+			Select(
+				key_fieldLabel = "Payment Status",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Completed");
+
+			Button.click(button = "Submit");
+
+			Refresh();
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:374]: I believe that it's more useful in testcase:

```
Button.click(button = "Submit");

Alert.viewSuccessMessage();
```

**Review by andreefarias — CHANGES_REQUESTED** [review]
Hey @davisaints,

Good job with this test. I request only some small changes! Thanks!

**Comment by davisaints** [comment]
ci:test:sf

**Review by andreefarias — APPROVED** [review]

**Comment by andreefarias** [comment]
Looks good to me! 🎉

**Comment by evanilson-santana** [comment]
ci:forward

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 5]
```diff
@@ -272,6 +272,116 @@ definition {
...
+			ApplicationsMenu.gotoSite(site = "Minium");
+
+			CommerceNavigator.gotoMiniumSidebarItem(menuItem = "Account Management");
+
+			CommerceAcceleratorsInitializer.createNewAccountViaFront(
+				accountName = "Commerce Account",
+				adminEmailAddress = "test@liferay.com");
+		}
+
+		task ("When: The user adds something to the cart and submits the order") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:324]: `The user adds an item to the cart and submits the order`

Just wanted to be a little more specific here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 6]
```diff
@@ -272,6 +272,116 @@ definition {
...
+			SelectFrame.selectFrameNoLoading(locator1 = "IFrame#MODAL_BODY");
+
+			Select(
+				key_fieldLabel = "Payment Status",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Completed");
+
+			Button.click(button = "Submit");
+
+			Alert.viewSuccessMessage();
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:375]: Same thing here, is there not a macro for this or at least some of it?

**Review by timpak — CHANGES_REQUESTED** [review]
Hey @davisaints, good job with this test, it's pretty complex! Can you take a look at my comments and get back to me? The wording needs to be updated for the first When.

For the other two, take a look at the Commerce tests and see how their tests are executing these steps. If they don't have macros, that's fine. If they do, we should try and use them as long as it makes sense for the test.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 7]
```diff
@@ -272,6 +272,116 @@ definition {
...
+				newAddressName = "Address Name",
+				newAddressZipCode = "Test Zip Code");
+
+			Click(locator1 = "Button#CONTINUE");
+
+			Click(
+				key_radioOption = "Standard",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
+
+			Click(locator1 = "Button#CONTINUE");
```
[the remark this answers is not in the document]
Comment (davisaints) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:350]: For this I found. Changes being made...

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
> Hey @davisaints, good job with this test, it's pretty complex! Can you take a look at my comments and get back to me? The wording needs to be updated for the first When.
> 
> For the other two, take a look at the Commerce tests and see how their tests are executing these steps. If they don't have macros, that's fine. If they do, we should try and use them as long as it makes sense for the test.

Hey @timpak! The changes were made and the delay in submitting for review was a bug involving actions (which has since been fixed). And an observation is that I didn't find the macros in Commerce, but I still managed to simplify this test.

**Comment by davisaints** [comment]
> Hey @davisaints, good job with this test, it's pretty complex! Can you take a look at my comments and get back to me? The wording needs to be updated for the first When.
> 
> For the other two, take a look at the Commerce tests and see how their tests are executing these steps. If they don't have macros, that's fine. If they do, we should try and use them as long as it makes sense for the test.

Hey @timpak! The changes were made and the delay in submitting for review was a bug involving actions (which has since been fixed). And an observation is that I didn't find the macros in Commerce, but I still managed to simplify this test.

**Comment by epark18** [comment]
ci:test:sf

**Comment by epark18** [comment]
@davisaints This looks good to me! I resent here https://github.com/liferay-bpm-qa/liferay-portal/pull/483 with some very minor typo fixes.

**Pull request closed without merging**